### PR TITLE
feat(NETWORK_INFORMATION): add new token

### PR DESCRIPTION
**Pull request by thekiba**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [x] Feature
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: projects/common/src/tokens/network-information.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import {inject, InjectionToken} from '@angular/core';
+import {NAVIGATOR} from './navigator';
+
+export const NETWORK_INFORMATION = new InjectionToken<NetworkInformation | undefined>(
```
Comment (waterplea) [inline projects/common/src/tokens/network-information.ts:4]: I believe even if you return `undefined` from factory, injected value would be `null`. Maybe explicitly convert it to `null` here?

File: projects/common/tsconfig.lib.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
         "target": "es2015",
         "declaration": true,
         "inlineSources": true,
-        "types": [],
+        "types": ["network-information-types"],
```
Comment (waterplea) [inline projects/common/tsconfig.lib.json:8]: Should we also add it to package devDependencies?

**Comment by waterplea** [comment]
I wonder if it should be named NETWORK_INFORMATION or CONNECTION. @MarsiBarsi what do you think?

**Comment by thekiba** [comment]
@waterplea NETWORK_INFORMATION is more informative for me instead of CONNECTION 😁

Please check the pr again, I force pushed the changes

**Comment by MarsiBarsi** [comment]
both options are okay for me 👍

**Comment by thekiba** [comment]
@waterplea I have reverted the change of returns `undefined` instead of `null` in the token factory for the tests to pass

**Comment by waterplea** [comment]
@thekiba instead you can add a second test where you would mock `NAVIGATOR` token with an object missing `connection` property. Test would be: "Injects `null` in unsupported browsers"

**Comment by thekiba** [comment]
It's a success now 🙌

**Comment by waterplea** [comment]
All looks well now. I wonder if there's a way not to force everybody to install those types if they do not use this token 🤔

**Comment by MarsiBarsi** [comment]
@waterplea I think it is a good idea. Let's make the type with "TODO to remove" after adding TS support

**Comment by waterplea** [comment]
Took the liberty to do those changes, related to types bundling. Looks like it works fine now, I tried a pre-release version on my pet project. Thank you, @thekiba !

**Pull request merged**